worldedit-core/src/main/java/com/sk89q/worldedit/util/eventbus/AnnotatedSubscriberFinder.java

```diff
@@ -59,7 +62,15 @@
                         );
                     }
                     Class<?> eventType = parameterTypes[0];
-                    EventHandler handler = new MethodEventHandler(annotation.priority(), listener, method);
+                    MethodHandle handle;
+                    try {
+                        handle = MethodHandles.lookup().unreflect(method);
+                        handle = handle.asType(handle.type().generic().changeReturnType(void.class));
```

Comment:
```suggestion
                        handle = MethodHandles.lookup().unreflect(method);
                        handle = handle.asType(MethodType.methodType(void.class, Object.class));
```
or maybe even extract the method type to a constant and just do:
```suggestion
                        handle = MethodHandles.lookup().unreflect(method).asType(EVENT_HANDLER_TYPE);
```

Reply:
Also consider caching `MethodHandles.lookup()`, iirc it's expensive to construct. Or use `publicLookup()`, which is probably more appropriate.

Reply:
... Also I'd rather move the `asType` to `MethodHandleEventHandler`, to be closer to where the `invokeExact` that needs it is.